### DISCUSSION TITLE
`list[str]` is not handled correctly in `BaseModel` classes using `Query`

### DISCUSSION THREAD

**Discussion by banuni** [fastapi/fastapi#8544] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Query, FastAPI
from pydantic import BaseModel
from typing import Optional

app = FastAPI()
class P(BaseModel):
    regular_str: str = Query(None) 
    nested_list_param: list[str] = Query([]) # {"regular":["hello"],"with dep":{"regular_str":"foo","nested_list_param":[]}}
    #nested_list_param: Optional[list[str]] = Query(None) # {"regular":["hello"],"with dep":{"regular_str":"foo","nested_list_param":null}}
    #nested_list_param: list[str] = Query(...) # {"detail":[{"loc":["body"],"msg":"field required","type":"value_error.missing"}]}


@app.get("/temp")
def temp(my_list_param: list[str] = Query(...), more_params: P = Depends()):
    return ({"regular": my_list_param, "with dep": more_params.dict()})
```

### Description

* Open the browser and navigate to `/temp?my_list_param=hello&nested_list_param=world&regular_str=foo`
* expected output would be
```json
{"regular":["hello"],"with dep":{"regular_str":"foo","nested_list_param":["world"]}}
```
but as you can see, commented in the code, the Query() dependency failed to extract the params which are defined as a list in the P class. the different definitions I tried yielded different results, but none of them allowed me to get it as I wanted.

### Operating System

macOS

### FastAPI Version

0.71.0

### Python Version

Python 3.10.0

**Comment by rafsaf** [comment, marked as answer]
I personally never seen combo with pydantic model and query params nor i doubt i could find it in docs, it looks... weird but appears to work somehow, even if not exactly as in a way you want. My guess is you are looking for this syntax.

```python
from fastapi import Query, FastAPI, Depends

app = FastAPI()


def im_dependency(
    regular_str: str = Query(None), nested_list_param: list[str] = Query([])
):
    return {"regular_str": regular_str, "nested_list_param": nested_list_param}


@app.get("/temp")
def temp(my_list_param: list[str] = Query(...), more_params=Depends(im_dependency)):
    return {"regular": my_list_param, "with dep": more_params}
```
or similar, which works and is fully documented (You can use classes also if you like). Maybe your syntax makes sense and I'm not aware of smth but it looks more like a mistake.

**Comment by harunyasar** [comment]
@banuni There's a great discussion [here](https://github.com/tiangolo/fastapi/issues/318) over the issue you have mentioned about and more.
